hack/lib/verify-any.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script verifies whether generated files are up-to-date. The
+# parameters are the command and its parameters for generating files
+# in the work tree.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/../..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+kube::util::ensure_clean_working_dir
+
+# This sets up the environment, like GOCACHE, which keeps the worktree cleaner.
+kube::golang::setup_env
+
+_tmpdir="$(kube::realpath "$(mktemp -d -t "$(basename "$0").XXXXXX")")"
```

Comment:
What is $0 here - is it "verify-any.sh"?  That's not super useful.  Maybe you should check that `$1` is not empty and use that here instead?

Reply:
In the new function, it is now `"verify-generated-$(basename "$1" | sed -e 's/[^a-zA-Z][^a-zA-Z]*/-/g').XXXXXX")"`.

`$1` cannot be empty, it must be a function or command that is to be executed.